nntrainer/tensor/float_tensor.cpp

```diff
@@ -87,6 +87,21 @@ FloatTensor::FloatTensor(
   }
 }
 
+bool FloatTensor::operator==(const FloatTensor &rhs) const {
+  const float *_data = (float *)getData();
+  const float *_rdata = (float *)rhs.getData();
+  for (size_t i = 0; i < size(); ++i) {
+    /** not checking sign change is intentional to avoid float calculation
+     * errors around 0 */
+    if ((std::isnan(_data[i]) && !std::isnan(_rdata[i])) ||
```

Comment:
if both Nan case ??

Reply:
This condition is previously changed from `std::isnan(data[i]) || std::isnan(rdata[i])` in #1964. It states it is to pass if both tensor has a value.

Reply:
however, I doubt subtracting nan from nan will output zero.

Reply:
@SeoHyungjun, @baek2sm, @djeong20 and I discussed offline about this problem.
For the case that comparing the 2 tensors which has a nan value as a element then we cannot say that these 2 tensors are identical even though this element is located at same position .

Reply:
This would help for this.
https://docs.oracle.com/javase/specs/jls/se8/html/jls-4.html#jls-4.2.3

```
NaN is unordered, so:

The numerical comparison operators <, <=, >, and >= return false if either or both operands are NaN ([§15.20.1](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.20.1)).

The equality operator == returns false if either operand is NaN.

In particular, (x<y) == !(x>=y) will be false if x or y is NaN.

The inequality operator != returns true if either operand is NaN ([§15.21.1](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.21.1)).

In particular, x!=x is true if and only if x is NaN.
```

Reply:
Fixed to `std::isnan(data[i]) || std::isnan(rdata[i])`.